src/D2L.CodeStyle.Analyzers/Attributes.cs

```diff
@@ -24,6 +24,7 @@ internal static class Mutability {
 			internal static readonly RoslynAttribute Unaudited = new RoslynAttribute( "D2L.CodeStyle.Annotations.Mutability.UnauditedAttribute" );
 		}
 		internal static readonly RoslynAttribute Singleton = new RoslynAttribute( "D2L.LP.Extensibility.Activation.Domain.SingletonAttribute" );
+		internal static readonly RoslynAttribute WebRequest = new RoslynAttribute( "D2L.LP.Extensibility.Activation.Domain.WebRequestAttribute" );

```

Comment:
RE: the name
Maybe `[OnlyWebRequest]`? There are some types shared in the singleton and webrequest graph that are fine. `[WebRequest]` looks a lot like `[Singleton]` but we're adding semantics to it that I don't think would apply to all `[WebRequest]` types. I could be convinced otherwise though, I'll come over IRL.

RE: this attributes file
I think the stuff in this file and `RoslynAttribute` etc. are not very good and are perpetuating some bad behaviours. I'll bug you IRL about an alternative (which is simpler and doesn't involve any custom stuff.)